docs: add instructions to upgrade admission controller (#9372)
rook/rook: Documentation/ceph-upgrade.md

#### **Updates for optional resources**

* [Prometheus monitoring](ceph-monitoring.md): upgrade Prometheus RBAC resources with this command.
  ```sh
  kubectl apply -f deploy/examples/monitoring/rbac.yaml
  ```

> * [Admission controller](admission-controller-usage.md): The admission controller has been

Reviewer: Should this be part of step 3 where the operator is updated? This is inside step 1 where the RBAC and CRDs are updated.

Author: I think it's fine to do it here. It is an optional resource, and so there is already a section for it. A user **could** get mixed up an accidentally skip step 2, but that step is also optional of a sort, so I'm not terribly worried about it breaking things.

kubectl -n $ROOK_OPERATOR_NAMESPACE set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.8.0
```

#### Admission controller

> At the same time, if you use the optional [Admission controller](admission-controller-usage.md),

Reviewer: ```suggestion
If you enabled the optional [Admission controller](admission-controller-usage.md),
```

there are additional updates. The admission controller has been integrated inside the operator
instead of a separate deployment. This means that the webhook server certificates are now stored in
the operator, and the operator manifest must be updated to use the one provided in

> `deploy/examples/operator.yaml`. If you are using Helm to manage the deployment there is nothing to

Reviewer: ```suggestion
`deploy/examples/operator.yaml`. If you are using Helm to manage the deployment this handled automatically.
```

do.

When updating the operator deployment with the latest example from Rook, there is some risk of

> overwriting changes you have made to the deployment or to the `rook-ceph-operator-config` ConfigMap.

Reviewer: ```suggestion
overwriting changes if you have customized the operator deployment or the `rook-ceph-operator-config` ConfigMap.
```
